src/main.zig

```diff
@@ -11,8 +11,8 @@ const enableRawMode = term.enableRawMode;
 const disableRawMode = term.disableRawMode;
 const getColumns = term.getColumns;
 
-pub const HintsCallback = (fn (Allocator, []const u8) Allocator.Error!?[]const u8);
-pub const CompletionsCallback = (fn (Allocator, []const u8) Allocator.Error![]const []const u8);
+pub const HintsCallback = *fn (Allocator, []const u8) Allocator.Error!?[]const u8;
+pub const CompletionsCallback = *fn (Allocator, []const u8) Allocator.Error![]const []const u8;
```

Comment:
The stage1 compiler requires the callbacks to be functions, not a function pointers.
These are referenced in line 152 and 227 in the `fun` variable. Currently, it is called like this:
```zig
try fun(self.allocator, self.buf.items);
```
Stage 1 can not dereference automatically, so a slight modification is required:
```zig
try fun.*(self.allocator, self.buf.items);
```
To support conditional compilation, depending on the compiler backend, it should be called like this:
```zig
const builtin = @import("builtin");
try (if (builtin.zig_backend == .stage1) fun.* else fun)(self.allocator, self.buf.items);
```